web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -308,41 +308,164 @@ odoo.define('web_timeline.TimelineRenderer', function (require) {
+            };
+        },
+        updateGroups: function(events) {
+            var self = this;
+            var group_ids = [];
+            // Get the missing groups from new events
+            _.each(events, function (event) {
+                var group = event[_.first(self.last_group_bys)];
+                if (group) {
+                    _.each(group, function (gr) {
+                        var found = _.find(self.groups, function(existing_group) {
+                            return existing_group.id == gr
+                        });
+                        if (found === undefined) {
+                            group_ids.push(gr);
+                        }
+                    });
+                }
+            });
+            var group_by_field = this.view.fields[self.last_group_bys];
+            self._rpc({
+                model: group_by_field.relation,
                 method: 'name_get',
                 args: [
-                    ids,
+                    group_ids,
                 ],
-                context: this.getSession().user_context,
+                context: self.getSession().user_context,
             }).then(function(names) {
-                var nevents = _.map(events, function (event) {
-                    return _.extend({
-                        __name: _.detect(names, function (name) {
-                            return name[0] === event.id;
-                        })[1]
-                    }, event);
+                _.each(names, function (gr_name) {
+                    self.groups.push({
+                        id: gr_name[0],
+                        content: gr_name[1],
+                    })
+                    self.$select_groups.append($("<option value='" + gr_name[0] + "' selected='selected'>" + gr_name[1] + "</option>"));
                 });
-                return self.on_data_loaded_2(nevents, group_bys, adjust_window);
+                self.$select_groups.multiselect('rebuild');
+                self.timeline.setGroups();
+                self.updateSelectedGroups();
+                return true;
+            });
+        },
+        arrayRemove: function (arr, id) {
+           return arr.filter( function (elmt) {
+               return elmt.id != id;
+           });
+        },
+        updateSelectedGroups: function() {
+            var self = this;
+            var selected = self.$select_groups.find("option:selected");
+            self.selected_groups = [];
+            _.each(selected, function (elmt) {
+                self.selected_groups.push({
+                    id: elmt.value,
+                    content: elmt.text,
+                })
+            });
+            self.timeline.setGroups(self.selected_groups);
+        },
+        createSelectGroups: function () {
+            var self = this;
+            var select_groups = $(core.qweb.render('TimelineSelectGroups', {'groups': self.groups}));
+            self.$el.find('.selected-groups').html(select_groups);
```

Comment:
@Tardo I don't see any good reason to change that.
The method find is more efficient than $(...) because $(...) is the method (jQuery) and does almost everything. This find is limited by the scope of $el (which is already shorter than the whole page).

Reply:
Odoo widgets have the "$" function defined that call to "this.$el.find()".

`this.$() != $()`